### DISCUSSION TITLE
Shorthand for prisma transaction

### DISCUSSION THREAD

**Discussion by felinto-dev** [prisma/prisma#12060] [Q&A]
Is there a shorthand for doing ONE transaction?

Instead to do it:

```ts
    await this.prisma.$transaction([
      this.prisma.hoster.update({
        where: { id: hosterId },
        data: { releaseAt: newReleaseAt },
      }),
    ]);
```

I would to do:

```ts
    await this.prisma.$transaction( <-- no array
      this.prisma.hoster.update({
        where: { id: hosterId },
        data: { releaseAt: newReleaseAt },
      }),
    ); <-- no array
```

I receive the error:
```ts
Type 'Prisma__HosterClient<Hoster>' is missing the following properties from type 'PrismaPromise<any>[]': length, pop, push, concat, and 29 more.
```

Is there an alternative?

**Comment by rujhanArora** [comment]
What's the issue with providing an array? Just wondering

**Reply by felinto-dev** [reply]
It's about code readability. I expect an array to always refer to a collection of elements.

**Comment by janpio** [comment]
Why do you want to put one single query into an additional transaction?

**Reply by felinto-dev** [reply]
Sorry, I didn't really understand your question. The only way to use transactions in prisma is through prisma.$transaction, right?

**Reply by janpio** [reply]
Yes, but what is the benefit of using a transaction for a single query? What do you gain by doing that?

**Reply by felinto-dev** [reply]
It's really complicated to explain all the business logic of my app that makes me need to use transactions, but I need to do a "count" in a table because I have a "credit system".

There is a risk of two NodeJS processes starting at the same time due to the microservices structure and therefore having concurrency issues. So I need to use a transaction for a single query.

**Reply by janpio** [reply]
I don't think there is a functional difference of sending a query directly or sending 1 query inside a transaction. If you need to avoid changes to a table between you opening the transaction and sending a query you would need to lock the table in the transaction as well, and thus have a multi query transaction.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
